.travis.yml

```diff
@@ -1,13 +1,23 @@
 sudo: required
 services: docker
-script: docker build . -t cs50/baseimage:${TRAVIS_BRANCH/master/latest}
+script: |
+  set -eo pipefail
+  [[ "$TRAVIS_BRANCH" == "master" ]] && TAG="latest" || TAG="$TRAVIS_BRANCH"
+  IMAGE="$TRAVIS_REPO_SLUG:$TAG"
+  docker build . -t "$IMAGE"
 before_deploy: docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
 deploy:
   provider: script
-  script: docker push cs50/baseimage:${TRAVIS_BRANCH/master/latest}
+  script: docker push "$IMAGE"
   on:
     all_branches: true
```

Comment:
Do we really want to push all branches, @kzidane?

Reply:
@dmalan this comes in handy when we push tags such as `python3.8`, etc so we don't have to change `.travis.yml` every time.